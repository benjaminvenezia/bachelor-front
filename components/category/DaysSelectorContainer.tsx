import { View, StyleSheet } from "react-native";
import DaySelector from "./DaySelector";
import DAYS from "../../constants/days";
import Title from "../ui/Title";

const daysLabels: Array<string> = [DAYS.MONDAY, DAYS.THURSDAY, DAYS.WEDNESDAY, DAYS.TUESDAY, DAYS.FRIDAY, DAYS.SATURDAY, DAYS.SUNDAY];

const DaysSelectorContainer = () => {
  return (
    <View>
      <Title titleType="h3">À quel(s) jour(s) souhaitez vous attribuer ces tâches?</Title>
      <View style={styles.container}>
        {daysLabels.map((day, index) => (
          <DaySelector key={index} label={day} />
        ))}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: "row",
    justifyContent: "space-around",
    flexWrap: "wrap",
    width: "100%",
  },
});

export default DaysSelectorContainer;
