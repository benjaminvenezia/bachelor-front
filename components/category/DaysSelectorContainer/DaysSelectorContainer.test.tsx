import React from "react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import DaysSelectorContainer from "./DaysSelectorContainer";
import configureStore from "redux-mock-store";

const mockStore = configureStore();

describe("<DaySelector />", () => {
  let store: any;

  beforeEach(() => {
    store = mockStore({ daysToAddTasks: { activeDays: [] } });
  });

  test("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <DaysSelectorContainer />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
