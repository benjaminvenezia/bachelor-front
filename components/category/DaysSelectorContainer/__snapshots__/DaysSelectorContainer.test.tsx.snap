// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DaySelector /> renders correctly 1`] = `
<View>
  <Text
    style={
      Array [
        Object {
          "fontWeight": "bold",
          "marginBottom": 10,
        },
        Object {
          "color": "#ffffff",
          "fontSize": 22,
        },
        undefined,
      ]
    }
  >
    À quel(s) jour(s) souhaitez vous attribuer ces tâches?
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "space-around",
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "borderColor": "white",
            "borderRadius": 100,
            "borderWidth": 5,
            "marginHorizontal": 3,
            "marginVertical": 5,
            "paddingHorizontal": 40,
            "paddingVertical": 15,
          },
          null,
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 20,
              },
            ]
          }
        >
          Lun
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "borderColor": "white",
            "borderRadius": 100,
            "borderWidth": 5,
            "marginHorizontal": 3,
            "marginVertical": 5,
            "paddingHorizontal": 40,
            "paddingVertical": 15,
          },
          null,
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 20,
              },
            ]
          }
        >
          Jeu
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "borderColor": "white",
            "borderRadius": 100,
            "borderWidth": 5,
            "marginHorizontal": 3,
            "marginVertical": 5,
            "paddingHorizontal": 40,
            "paddingVertical": 15,
          },
          null,
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 20,
              },
            ]
          }
        >
          Mer
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "borderColor": "white",
            "borderRadius": 100,
            "borderWidth": 5,
            "marginHorizontal": 3,
            "marginVertical": 5,
            "paddingHorizontal": 40,
            "paddingVertical": 15,
          },
          null,
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 20,
              },
            ]
          }
        >
          Mar
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "borderColor": "white",
            "borderRadius": 100,
            "borderWidth": 5,
            "marginHorizontal": 3,
            "marginVertical": 5,
            "paddingHorizontal": 40,
            "paddingVertical": 15,
          },
          null,
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 20,
              },
            ]
          }
        >
          Ven
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "borderColor": "white",
            "borderRadius": 100,
            "borderWidth": 5,
            "marginHorizontal": 3,
            "marginVertical": 5,
            "paddingHorizontal": 40,
            "paddingVertical": 15,
          },
          null,
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 20,
              },
            ]
          }
        >
          Sam
        </Text>
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "borderColor": "white",
            "borderRadius": 100,
            "borderWidth": 5,
            "marginHorizontal": 3,
            "marginVertical": 5,
            "paddingHorizontal": 40,
            "paddingVertical": 15,
          },
          null,
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 20,
              },
            ]
          }
        >
          Dim
        </Text>
      </View>
    </View>
  </View>
</View>
`;
